network/validator/pubsub/authorized_sender_validator_test.go

```diff
@@ -0,0 +1,293 @@
+// validation errors for all possible valid combinations (authorized sender role, message type).
+func (s *TestAuthorizedSenderValidatorSuite) TestValidatorCallback_AuthorizedSender() {
+	for _, c := range s.authorizedSenderTestCases {
+		str := fmt.Sprintf("role (%s) should be authorized to send message type (%s) on channel (%s)", c.Identity.Role, c.MessageStr, c.Channel)
+		s.Run(str, func() {
+			validate := AuthorizedSenderValidator(zerolog.Nop(), c.Channel, c.GetIdentity)
+
+			pid, err := unittest.PeerIDFromFlowID(c.Identity)
+			s.Require().NoError(err)
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			msgType, err := validate(ctx, pid, c.Message)
+			s.Require().NoError(err)
+			s.Require().Equal(c.MessageStr, msgType)
+
+			validatePubsub := AuthorizedSenderMessageValidator(zerolog.Nop(), c.Channel, c.GetIdentity)
+			pubsubResult := validatePubsub(ctx, pid, c.Message)
+			s.Require().Equal(pubsub.ValidationAccept, pubsubResult)
+		})
+	}
+}
+
+// TestValidatorCallback_UnAuthorizedSender checks that the call back returned from AuthorizedSenderValidator return's ErrUnauthorizedSender
+// validation error for all possible invalid combinations (unauthorized sender role, message type).
+func (s *TestAuthorizedSenderValidatorSuite) TestValidatorCallback_UnAuthorizedSender() {
+	for _, c := range s.unauthorizedSenderTestCases {
+		str := fmt.Sprintf("role (%s) should not be authorized to send message type (%s) on channel (%s)", c.Identity.Role, c.MessageStr, c.Channel)
+		s.Run(str, func() {
+			pid, err := unittest.PeerIDFromFlowID(c.Identity)
+			s.Require().NoError(err)
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			validate := AuthorizedSenderValidator(zerolog.Nop(), c.Channel, c.GetIdentity)
+			msgType, err := validate(ctx, pid, c.Message)
+			s.Require().ErrorIs(err, message.ErrUnauthorizedRole)
+			s.Require().Equal(c.MessageStr, msgType)
+
+			validatePubsub := AuthorizedSenderMessageValidator(zerolog.Nop(), c.Channel, c.GetIdentity)
+			pubsubResult := validatePubsub(ctx, pid, c.Message)
+			s.Require().Equal(pubsub.ValidationReject, pubsubResult)
+		})
+	}
+}
+
+// TestValidatorCallback_UnAuthorizedMessageOnChannel for each invalid combination of message type and channel
+// the call back returned from AuthorizedSenderValidator returns the appropriate error message.ErrUnauthorizedMessageOnChannel.
+func (s *TestAuthorizedSenderValidatorSuite) TestValidatorCallback_UnAuthorizedMessageOnChannel() {
+	for _, c := range s.unauthorizedMessageOnChannelTestCases {
+		str := fmt.Sprintf("message type (%s) should not be authorized to be sent on channel (%s)", c.MessageStr, c.Channel)
+		s.Run(str, func() {
+			pid, err := unittest.PeerIDFromFlowID(c.Identity)
+			s.Require().NoError(err)
+
+			ctx, cancel := context.WithCancel(context.Background())
```

Comment:
nit: You can just create a single context for the whole test method instead of creating it in the loop